### DISCUSSION TITLE
Crossplane pods are failed with TLS handshake error

### DISCUSSION THREAD

**Discussion by balajikandan** [crossplane/crossplane#5056] [Q&A] [answered]
Crossplane pods are running and throws below error. The issue does not happen if I point and pull image from docker hub

`Failed to pull image "xpkg.upbound.io/crossplane/crossplane:v1.14.0": rpc error: code = Unknown desc = failed to pull and unpack image "xpkg.upbound.io/crossplane/crossplane:v1.14.0": failed to resolve reference "xpkg.upbound.io/crossplane/crossplane:v1.14.0": failed to do request: Head "https://xpkg.upbound.io/v2/crossplane/crossplane/manifests/v1.14.0": net/http: TLS handshake timeout`

**Comment by jbw976** [comment]
Is this consistently happening for you @balajikandan, or did you notice it to be temporary/transient?

**Reply by balajikandan** [reply]
@jbw976  Form my local laptop crossplane working without any issue, but from EKS AWS environment. I am seeing this issue.

**Reply by jbw976** [reply]
Thanks for confirming that @balajikandan. A couple more questions:

* is this a plain vanilla EKS cluster, or is there any additional config already added that is worth mentioning here?
* Do you know of any config options (network policies, security groups, egress rules, etc.) that could potentially be updated on your EKS cluster to enable access from it to `xpkg.upbound.io`?

I'm a bit worried about this affecting more folks over time, so I'm going to convert it to an issue now for better tracking ✅

**Reply by jbw976** [reply]
Tracking issue: https://github.com/crossplane/crossplane/issues/5065

**Reply by balajikandan** [reply]
It is a plain vanilla EKS cluster, From a security and firewall perspective HTTP port 443 is opened to the upbound.io domain. Is there any document available that provides steps we need to follow to whitelist to connect the upbound docker hub ?

**Reply by jastang** [reply]
Hi @balajikandan -- just wanted to confirm that your firewall configuration hasn't changed since the last message. Does it default to denying all traffic and opening 443 is the only edit you've made?

**Reply by balajikandan** [reply]
@jastang That is correct, for crossplane we have opened 443 for upbound.io domains

**Reply by thephred** [reply]
The domains we can see being used for a pull are `xpkg.upbound.io` and `d3qrbvrml4iuq4.cloudfront.net`. Could you verify that with those allowed it works?

**Reply by balajikandan** [reply]
Currently xpkg.upbound.io is open, still seeing the issue mentioned. Do I need to exclude this domain too d3qrbvrml4iuq4.cloudfront.net  ?

**Reply by thephred** [reply]
Yes, that is where the actual oci layers are pulled from.

**Reply by balajikandan** [reply]
We have whitelisted both the domains. Still we are facing the same issue. TLS handshake is failing with no CA or client certs available. 

From EKS cluster, we ran this command to check HTTP connectivity. It looks like connectivity exists but the SSL certs are not available. 

~ openssl s_client -connect xpkg.upbound.io:443 -tls1_2

CONNECTED(00000003)
write:errno=0

no peer certificate available

No client certificate CA names sent

SSL handshake has read 0 bytes and written 218 bytes
Verification: OK

New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID:
    Session-ID-ctx:
    Master-Key:
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1706046183
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no


Am I missing something here ?

**Comment by balajikandan** [comment, marked as answer]
It is a Operating system issue, Required CA certs were missing. We are able to download the images from the xpkg.upbound.io

**Reply by jbw976** [reply]
Thank you very much for following up on this @balajikandan to help explain the root cause. If there's any other description you can provide about how you installed the required CA certs, feel free to share them here as they may prove useful to others in the future 🙏
